x-pack/filebeat/fbreceiver/receiver_test.go

```diff
@@ -248,6 +249,51 @@ func TestMultipleReceivers(t *testing.T) {
 	})
 }
 
+func TestStatusReporter(t *testing.T) {
+	cfg := Config{
+		Beatconfig: map[string]any{
+			"filebeat": map[string]any{
+				"inputs": []map[string]any{
+					{
+						"type":    "benchmark",
+						"enabled": true,
+						"message": "test",
+						"count":   10,
+					},
+				},
+			},
+			"output": map[string]any{
+				"otelconsumer": map[string]any{},
+			},
+			"logging": map[string]any{
+				"level": "info",
+				"selectors": []string{
+					"*",
+				},
+			},
+			"path.home": t.TempDir(),
+		},
+	}
+
+	host := &oteltest.MockHost{}
+	oteltest.CheckReceivers(oteltest.CheckReceiversParams{
+		T:    t,
+		Host: host,
+		Receivers: []oteltest.ReceiverConfig{
+			{
+				Name:    "r1",
+				Config:  &cfg,
+				Factory: NewFactory(),
+			},
+		},
+		AssertFunc: func(c *assert.CollectT, _ map[string][]mapstr.M, _ *observer.ObservedLogs) {
+			require.NotNil(t, host.Evt)
+			require.Nil(t, host.Evt.Err())
+			require.Equal(t, host.Evt.Status(), componentstatus.StatusOK)
+		},
```

Comment:
Can we add this to TestNewReceiver? I believe the only difference is that here we pass a host and make assertions, you could do the same in TestNewReceiver. Each test here takes 15 seconds to run, so if we can avoid making that worse, it's a win.